lti_consumer/plugin/views.py

```diff
@@ -44,15 +49,22 @@ def public_keyset_endpoint(request, usage_id=None):
     and run the proper handler.
     """
     try:
-        usage_key = UsageKey.from_string(usage_id)
-
-        return run_xblock_handler_noauth(
-            request=request,
-            course_id=str(usage_key.course_key),
-            usage_id=str(usage_key),
-            handler='public_keyset_endpoint'
+        lti_config = LtiConfiguration.objects.get(
+            location=UsageKey.from_string(usage_id)
         )
-    except Exception:  # pylint: disable=broad-except
+
+        if lti_config.version != lti_config.LTI_1P3:
+            raise LtiError(
+                "LTI Error: LTI 1.1 blocks do not have a public keyset endpoint."
+            )
+
+        # Retrieve block's Public JWK
+        # The underlying method will generate a new Private-Public Pair if one does
+        # not exist, and retrieve the values.
+        response = JsonResponse(json.loads(lti_config.lti_1p3_platform_public_jwk))
```

Comment:
Decoding and then re-encoding the JSON seems unfortunate, but I guess it's easier to use JsonResponse like this...

Reply:
I've changed it a bit, now the model returns a proper python object that can be passed directly to `JsonResponse`